core/src/main/java/org/apache/iceberg/io/PartitioningWriterFactory.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.deletes.PositionDelete;
+import org.apache.iceberg.io.DefaultPartitioningWriterFactory.Type;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public interface PartitioningWriterFactory<T> {
```

Comment:
Why name it as `PartitioningWriterFactory` ?  I don't see any partition info for those defined interface methods.

Reply:
It's factory class used to create `PartitioningWriter`s

Reply:
I just add a little docs for better understanding.